bst/bst.py

```diff
@@ -0,0 +1,112 @@
+# -*- coding: utf-8 -*-
+from __future__ import unicode_literals
+from time import time
+
+
+class Node():
+    def __init__(self, value):
+        self.value = value
+        self.left = None
+        self.right = None
+
+
+class Tree():
+    def __init__(self):
+        self._head = None
+        self._size = 0
+        self._depth = 0
+        self._balance = 0
+
+    def insert(self, value):
+        """insert value into tree. if already present, ignored"""
+        n = Node(value)
+        if self._head is None:
+            self._head = n
+            self._size += 1
+            self._depth += 1
+            return
+
+        parent = None
+        direction = 'left'
+        current = self._head
+        depth = 1
+
+        if value < self._head.value:
+            self._balance += 1
+        elif value > self._head.value:
+            self._balance -= 1
```

Comment:
this will not always be true.  Balance does not change in the case that the new value becomes the second child of a node that already has one child.